base/src/dispatcher/dispatcher.ts

```diff
@@ -555,11 +559,11 @@ export class Dispatcher implements ZLUX.Dispatcher {
   invokeAction(action:Action, eventContext: any):any{
     this.log.info("dispatcher.invokeAction on context "+JSON.stringify(eventContext));
     this.getActionTarget(action,eventContext).then( (target: ActionTarget) => {
-      const wrapper = target.wrapper;
+      const wrapper = target.wrapper; 
+      var  targetId = eventContext.data.target || this.windowManager.nextId() - 1;
```

Comment:
This should not be done for various reasons:
1. eventContext is given to the consumer, so the app would see data.target and be confused about its meaning. If targetId is needed, it should be given as a 3rd parameter to invokeAction.
2. I believe the IDs should be made randomized to make guessing IDs harder. You should assume you cannot know the ID, and if one isnt provided then the dispatcher should not be able to do an action for a specific instance.
3. never var, always let or const because var's scope is not localized. It is strange.